src/bp-core/bp-core-update.php

```diff
@@ -3226,3 +3230,14 @@ function bb_update_to_2_4_10() {
 		}
 	}
 }
+
+/**
+ * For existing install disable pin post setting by default.
+ * 
+ * @since BuddyBoss [BBVERSION]
+ *
+ * @return void
+ */
+function bb_update_to_2_4_50() {
+	bp_update_option( '_bb_enable_activity_pinned_posts', 0 );
```

Comment:
Make this default enable for the fresh install in the function `bp_get_default_options()`